python_logging_rabbitmq/handlers_oneway.py

```diff
@@ -150,20 +151,23 @@ def message_worker(self):
             try:
                 record, routing_key = self.queue.get(block=True, timeout=10)
 
-                if not self.connection or self.connection.is_closed or not self.channel or self.channel.is_closed:
-                    self.open_connection()
-
-                res = self.channel.basic_publish(
-                    exchange=self.exchange,
-                    routing_key=routing_key,
-                    body=record,
-                    properties=pika.BasicProperties(
-                        delivery_mode=2,
-                        headers=self.message_headers
+                try:
+                    if not self.connection or self.connection.is_closed or not self.channel or self.channel.is_closed:
+                        self.open_connection()
+
+                    res = self.channel.basic_publish(
+                        exchange=self.exchange,
+                        routing_key=routing_key,
+                        body=record,
+                        properties=pika.BasicProperties(
+                            delivery_mode=2,
+                            headers=self.message_headers
+                        )
                     )
-                )
+                finally:
+                    self.queue.task_done()
```

Comment:
We be here if and only if a message was dequeued. So, this is the proper place to call task_done.